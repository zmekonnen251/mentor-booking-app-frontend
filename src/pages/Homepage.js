import { useSelector } from 'react-redux';
import Mentors from '../components/Mentors';
import style from './Homepage.module.css';

export default function Homepage() {
  const approvedMentors = useSelector((state) => state.mentors.approvedMentors);
  const mentors = approvedMentors?.filter((mentor) => mentor.role !== 'superadmin');

  return (
    <div className={style.homepage}>
      <div className={style.pagetitle}>
        <h1 className={style.maintitle}>
          Our Mentors
        </h1>
        <p>Please select a mentor</p>
      </div>
      <Mentors mentors={mentors} />
    </div>
  );
}
