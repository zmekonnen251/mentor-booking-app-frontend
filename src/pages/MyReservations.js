/* eslint-disable react/no-unknown-property */
import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { TailSpin } from 'react-loading-icons';
import style from './MyReservations.module.css';
import { cancelReservation } from '../redux/actions/mentorReservation';

export default function MyReservations() {
  const dispatch = useDispatch();
  const reservations = useSelector((state) => state.reservations);

  return (
    <div className={style.myreservations}>
      <div className={style.pagetitle}>
        <h1 className={style.maintitle}>
          My Reservations
        </h1>
        <p>List of All reserved sessions with reservations</p>
      </div>

      <div className={style.mentorstable}>
        <table>
          <thead>
            <tr>
              <th>Mentors</th>
              <th className={style.datecolumn}>Scheduled Date</th>
              <th>Actions</th>
            </tr>
          </thead>
          <tbody>
            {reservations.length === 0 ? <TailSpin />
              : reservations.map((reservation) => (
                <tr key={reservation.id}>
                  <td className={style.mentordata}>
                    <img className={style.avatar} src={reservation.avatar} alt={`${reservation.mentor_name}`} />
                    <div className={style.mentorname}>
                      <h3>{reservation.mentor_name}</h3>
                      <h3>{reservation.mentor_email}</h3>
                      <p>{reservation.city}</p>
                      <p>{reservation.country}</p>
                    </div>
                  </td>
                  <td className={style.datecolumn}>{reservation.date}</td>
                  <td className="hr-center">
                    <button
                      type="button"
                      className={style.deletebtn}
                      onClick={() => dispatch(cancelReservation(reservation.id))}
                    >
                      Cancel Reservation
                    </button>
                  </td>
                </tr>
              ))}
          </tbody>
        </table>
      </div>
    </div>
  );
}
