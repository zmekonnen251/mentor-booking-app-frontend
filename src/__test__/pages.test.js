import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from '../redux/store';
import Homepage from '../pages/Homepage';
import MyReservations from '../pages/MyReservations';

describe('Testing if all pages render correctly', () => {
  it('Testing Homepage page', () => {
    const tree = render(
      <Provider store={configureStore}>
        <Router>
          <Homepage />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Testing MyReservations page', () => {
    const tree = render(
      <Provider store={configureStore}>
        <Router>
          <MyReservations />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
