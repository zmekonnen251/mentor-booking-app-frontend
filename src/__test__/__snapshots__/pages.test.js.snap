// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if all pages render correctly Testing Homepage page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="homepage"
      >
        <div
          class="pagetitle"
        >
          <h1
            class="maintitle"
          >
            Our Mentors
          </h1>
          <p>
            Please select a mentor
          </p>
        </div>
        <ul
          class="mentors"
        >
          <svg
            class="icon-loading"
            height="38"
            viewBox="0 0 38 38"
            width="38"
          >
            <defs>
              <lineargradient
                id="prefix__a"
                x1="8.042%"
                x2="65.682%"
                y1="0%"
                y2="23.865%"
              >
                <stop
                  offset="0%"
                  stop-opacity="0"
                />
                <stop
                  offset="63.146%"
                  stop-opacity="0.631"
                />
                <stop
                  offset="100%"
                />
              </lineargradient>
            </defs>
            <g
              fill="none"
              fill-rule="evenodd"
              transform="translate(1 1)"
            >
              <path
                d="M36 18c0-9.94-8.06-18-18-18"
                stroke="#97bf0f"
                stroke-width="3"
              >
                <animatetransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </path>
              <circle
                cx="36"
                cy="18"
                r="1"
              >
                <animatetransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </circle>
            </g>
          </svg>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="homepage"
    >
      <div
        class="pagetitle"
      >
        <h1
          class="maintitle"
        >
          Our Mentors
        </h1>
        <p>
          Please select a mentor
        </p>
      </div>
      <ul
        class="mentors"
      >
        <svg
          class="icon-loading"
          height="38"
          viewBox="0 0 38 38"
          width="38"
        >
          <defs>
            <lineargradient
              id="prefix__a"
              x1="8.042%"
              x2="65.682%"
              y1="0%"
              y2="23.865%"
            >
              <stop
                offset="0%"
                stop-opacity="0"
              />
              <stop
                offset="63.146%"
                stop-opacity="0.631"
              />
              <stop
                offset="100%"
              />
            </lineargradient>
          </defs>
          <g
            fill="none"
            fill-rule="evenodd"
            transform="translate(1 1)"
          >
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
              stroke="#97bf0f"
              stroke-width="3"
            >
              <animatetransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
            <circle
              cx="36"
              cy="18"
              r="1"
            >
              <animatetransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </circle>
          </g>
        </svg>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing if all pages render correctly Testing MyReservations page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="myreservations"
      >
        <div
          class="pagetitle"
        >
          <h1
            class="maintitle"
          >
            My Reservations
          </h1>
          <p>
            List of All reserved sessions with reservations
          </p>
        </div>
        <div
          class="mentorstable"
        >
          <table>
            <thead>
              <tr>
                <th>
                  Mentors
                </th>
                <th
                  class="datecolumn"
                >
                  Scheduled Date
                </th>
                <th>
                  Actions
                </th>
              </tr>
            </thead>
            <tbody>
              <svg
                class="icon-loading"
                height="38"
                viewBox="0 0 38 38"
                width="38"
              >
                <defs>
                  <lineargradient
                    id="prefix__a"
                    x1="8.042%"
                    x2="65.682%"
                    y1="0%"
                    y2="23.865%"
                  >
                    <stop
                      offset="0%"
                      stop-opacity="0"
                    />
                    <stop
                      offset="63.146%"
                      stop-opacity="0.631"
                    />
                    <stop
                      offset="100%"
                    />
                  </lineargradient>
                </defs>
                <g
                  fill="none"
                  fill-rule="evenodd"
                  transform="translate(1 1)"
                >
                  <path
                    d="M36 18c0-9.94-8.06-18-18-18"
                    stroke="#fff"
                    stroke-width="2"
                  >
                    <animatetransform
                      attributeName="transform"
                      dur="0.9s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </path>
                  <circle
                    cx="36"
                    cy="18"
                    r="1"
                  >
                    <animatetransform
                      attributeName="transform"
                      dur="0.9s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </circle>
                </g>
              </svg>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="myreservations"
    >
      <div
        class="pagetitle"
      >
        <h1
          class="maintitle"
        >
          My Reservations
        </h1>
        <p>
          List of All reserved sessions with reservations
        </p>
      </div>
      <div
        class="mentorstable"
      >
        <table>
          <thead>
            <tr>
              <th>
                Mentors
              </th>
              <th
                class="datecolumn"
              >
                Scheduled Date
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody>
            <svg
              class="icon-loading"
              height="38"
              viewBox="0 0 38 38"
              width="38"
            >
              <defs>
                <lineargradient
                  id="prefix__a"
                  x1="8.042%"
                  x2="65.682%"
                  y1="0%"
                  y2="23.865%"
                >
                  <stop
                    offset="0%"
                    stop-opacity="0"
                  />
                  <stop
                    offset="63.146%"
                    stop-opacity="0.631"
                  />
                  <stop
                    offset="100%"
                  />
                </lineargradient>
              </defs>
              <g
                fill="none"
                fill-rule="evenodd"
                transform="translate(1 1)"
              >
                <path
                  d="M36 18c0-9.94-8.06-18-18-18"
                  stroke="#fff"
                  stroke-width="2"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="0.9s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </path>
                <circle
                  cx="36"
                  cy="18"
                  r="1"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="0.9s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </circle>
              </g>
            </svg>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
